pkg/reconciler/pingsource/controller/controller.go

```diff
@@ -83,11 +85,22 @@ func NewController(
 	r.Logger.Info("Setting up event handlers")
 	pingSourceInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
 
+	// Watch for deployments owned by the source
 	deploymentInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
 		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("PingSource")),
 		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
 	})
 
+	// Watch for the global deployment
+	gr := func(obj interface{}) {
+		impl.GlobalResync(pingSourceInformer.Informer())
```

Comment:
why do we need a global resync, maybe add a comment here.

Reply:
I assume it's because the shared deployment for multi-tenant ones will need to requeue all the ping sources, but if there's only one there, can we not just use tracker for it and only watch / notify on changes to that deployment?

Reply:
What's the difference between the tracker and the informer with a FilterFunc?

Reply:
Well, I'm not sure why we are doing a global resync here. with the tracker you know if it's using the shared one and you can set up the tracker to only queue the particular ping source when things change. Here you're saying (unless I'm mistaken) that whenever the shared deployment changes, enqueue all the ping sources regardless of if they are affected by the change or not.

Reply:
ah got it, thanks! You are not mistaken. 

This is related to the open discussion about whether to allow mixing cluster and namespace scoped dispatchers/jobrunners. Let's say we allow it (and offer config knobs to restrict it) then using the tracker is a win in that case. Working on it...